RealEstateDataCleaning.py

```diff
@@ -8,113 +8,160 @@
+    """Takes a list and makes a column for each phrase in that list
+    it places a 1 in that column if that phrase is present in the description else it places a 0
 
     Args:
         df (pandas DataFrame): The DataFrame containing the descriptions
         features (list): The list of phrases that are being looked for in descriptions
-        col (str): name of column which contains the descriptions  
+        col (str): name of column which contains the descriptions
 
     Returns:
-        pandas DataFrame: without the column which contains descriptions plus 
-        additional columns one for each word in features 
-        
-    '''
-    
+        pandas DataFrame: without the column which contains descriptions plus
+        additional columns one for each word in features
+
+    """
+
     for feature in features:
         df[feature] = np.where((df[col].str.contains(feature)),1,0)
     df.drop([col], axis=1, inplace=True)
     return df
 
-def main():
-	df = pd.read_csv(input_file)
-	df.drop(df[df.SqFtTotal == 0].index, inplace=True)
-	df['CloseDate']  = pd.to_datetime(df['CloseDate'])
-	df = concatenate_address(df)
-	df = extract_dummies(df, Exterior_features, col='Exterior')
-	df['BasementDesc'] = df['BasementDesc'].apply(Basement_cleanup)
-	df = extract_dummies(df, BasementDesc_features, col='BasementDesc')
-	df = extract_dummies(df, ParkingDesc_features, col='ParkingDesc')
-	df['Stories'] = df['Stories'].apply(Stories_cleanup) 
-	df = extract_dummies(df, Stories_features, col='Stories')
-	df = make_YearBuilt_bins(df, size=10)
-	df['Year'] = pd.DatetimeIndex(df['CloseDate']).year
-	df['Age'] = df['Year'] - df['YearBuilt']
-	df['AssociationFee'] = np.where((df['AssociationFee']> 0),1,0)
-	df['PoolonProperty'] = np.where((df['PoolonProperty'] !=  'None'),1,0)
-	df['Bedrooms'] = df['UpperBedrooms'] + df['MainBedrooms']
-	df['FullBaths'] = df['UpperFullBaths'] + df['MainFullBaths']
-	df['HalfBaths'] = df['UpperHalfBaths'] + df['MainHalfBaths']
-	df.drop(['StreetNumber', 'StreetName', 'City', 'CountyOrParish', 'StateOrProvince', 
-			 'PostalCode','UpperBedrooms', 'MainBedrooms', 'UpperFullBaths', 'MainFullBaths', 
-			 'UpperHalfBaths', 'MainHalfBaths'], axis=1, inplace=True)
-	df.to_csv(output_file)
+def add_upper_and_main(data_frame, desired):
+    """
+    Look for the "Upper" and "Main" values for the desired column.  Sum these together and
+    create a new column named desired.  Finally, remove the "Upper" and "Main" columns
+
+    e.g. "Bedrooms" is the sum of the "UpperBedrooms" and "MainBedrooms" values
+
+    :param data_frame: callback frame object to manipulate
+    :param desired: string denoting the desired domain
+    :return: None
+    """
+    upper = "Upper{}".format(desired)
+    main = "Main{}".format(desired)
+
+    data_frame[desired] = data_frame[upper] + data_frame[main]
+
+    data_frame.drop([upper, main], axis=1, inplace=True)
+
+def main(df):
+    # method does too much - change to be testable (see example)
```

Comment:
I did a little bit of refactoring here.  `main` is doing an awful lot of work.  The work should be split up a little bit.  So, for instance, this method doesn't care about the "csv" parts except for the very beginning and the very end.  You can pull out the "csv" but and just have this operate on the dataframe.